### PR TITLE
r.mess add check topology vector reference layer

### PR DESCRIPTION
**Pull request by ecodiv**
The input reference vector layer needs to be a point layer. Added a check. If a reference point layer is given, but it does not contain points, the r.mess will exit with an error message.

### CODE REVIEW

File: src/raster/r.mess/r.mess.py [hunk 1]
```diff
@@ -192,9 +193,19 @@ def main(options, flags):
         return 0
 
     # Reference / sample area or points
-    ref_rast = options["ref_rast"]
     ref_vect = options["ref_vect"]
-    if ref_rast:
+    if bool(ref_vect):
+        topology_check = gs.parse_command("v.info", map=ref_vect, flags="t")
```
Comment (petrasovaa) [inline src/raster/r.mess/r.mess.py]: There is [`gs.vector_info_topo`](https://grass.osgeo.org/grass83/manuals/libpython/script.html#script.vector.vector_info_topo) function you could use here.

File: src/raster/r.mess/r.mess.py [hunk 2]
```diff
@@ -192,9 +193,19 @@ def main(options, flags):
         return 0
 
     # Reference / sample area or points
-    ref_rast = options["ref_rast"]
     ref_vect = options["ref_vect"]
-    if ref_rast:
+    if bool(ref_vect):
```
Comment (petrasovaa) [inline src/raster/r.mess/r.mess.py]: I don't think the bool conversion is needed, it's automatic.

Comment (ecodiv) [reply src/raster/r.mess/r.mess.py]: OK, thanks. It was recommended to me once, but can't remember the specifics.

Comment (ecodiv) [reply src/raster/r.mess/r.mess.py]: Done

File: src/raster/r.mess/r.mess.py (hunk 1 above)
Comment (ecodiv) [reply src/raster/r.mess/r.mess.py]: Thanks, done.

**Pull request merged**